code/controllers/subsystem/mapping.dm

```diff
@@ -737,11 +741,25 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 	if(below_offset)
 		update_plane_tracking(new_z)
 
+	build_area_turfs(z_value, filled_with_space)
+
 	// And finally, misc global generation
 
 	// We'll have to update this if offsets change, because we load lowest z to highest z
 	generate_lighting_appearance_by_z(z_value)
 
+/datum/controller/subsystem/mapping/proc/build_area_turfs(z_level, space_guarenteed)
+	// If we know this is filled with default tiles, we can use the default area
+	// Faster
+	if(space_guarenteed)
+		var/area/global_area = GLOB.areas_by_type[world.area]
+		global_area.contained_turfs += Z_TURFS(z_level)
+		return
+
+	for(var/turf/lad as anything in Z_TURFS(z_level))
```

Comment:
real name

you are using "lad" both to mean area in one place and turf to mean other

this is just x but longer